React/Views/RCTView.m

```diff
@@ -341,6 +361,136 @@ - (NSString *)accessibilityValue
   }
   return nil;
 }
+#else // [TODO(macOS ISS#2323203)
+- (id)accessibilityValue {
+  if (self.accessibilityRole == NSAccessibilityCheckBoxRole) {
+    for (NSString *state in self.accessibilityState) {
+      id val = self.accessibilityState[state];
+      if (!val) {
+        continue;
+      }
+      if ([state isEqualToString:@"checked"]) {
+        if ([val isKindOfClass:[NSNumber class]]) {
+          return [val boolValue] ? @YES : @NO;
```

Comment:
I was because JSON bridged to ObjC, numbers and booleans just become NSNumber and VoiceOver treats `2` as "mixed" instead of "checked".   But someone would have had to write `{ state: 2 }` instead of the documented `{ state: 'mixed' }` which is the case handled below in the `else if`